qutip/qip/noise.py

```diff
@@ -504,7 +433,7 @@ class UserNoise(Noise):
     def __init__(self):
         pass
 
-    def get_noisy_dynamics(self, pulses, dims):
+    def get_noisy_dynamics(self, pulses, systematic_noise, dims):
```

Comment:
This doesn't match the signature of other `get_noisy_dynamics` calls in the inheritance chain.  If there's parsing to be done on all calls, you could always have `get_noisy_dynamics` to be defined in the base class, and ask subclasses to overide a related method `_get_noisy_dynamics`, which would have much stronger guarantees about exactly what will be passed in its arguments (or have them pass in that function to the `UserNoise` constructor so things are composed not overridden, but I just hate OO programming so ignore me).